Engine/ResourceManagement/Importer/StateMachineImporter.cpp

```diff
@@ -75,26 +67,20 @@ ImportResult StateMachineImporter::Import(const File & file, bool force) const
 
 		bytes = sizeof(uint64_t);
 		memcpy(cursor, &transition->target_hash, bytes);
-		cursor += bytes; 
+		cursor += bytes;
 
 		bytes = sizeof(uint64_t);
 		memcpy(cursor, &transition->trigger_hash, bytes);
-		cursor += bytes; 
+		cursor += bytes;
 
 		bytes = sizeof(uint64_t);
 		memcpy(cursor, &transition->interpolation_time, bytes);
-		cursor += bytes; 
+		cursor += bytes;
 	}
 
 	bytes = sizeof(uint64_t);
 	memcpy(cursor, &state_machine.default_state, bytes);
 	cursor += bytes;
 
-	std::string exported_file = SaveMetaFile(file.file_path, ResourceType::STATE_MACHINE);
-	App->filesystem->Save(exported_file.c_str(), data, size);
-
-	import_result.success = true;
-	import_result.exported_file = exported_file;
-	free(data);
-	return import_result;
+	return FileData{data, size};
```

Comment:
the data here where you make it free?

Reply:
This function is called from class `Importer`, in line 40 and 41:
```
FileData imported_data = ExtractData(assets_file_path, *metafile);
metafile_exported_folder_path->Save(std::to_string(metafile->uuid).c_str(), imported_data);
```
In line 41 the `Save` method takes this data, saves it in a file and also frees it